zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/VFSNotebookRepo.java

```diff
@@ -159,6 +159,7 @@ private Note getNote(FileObject noteDir) throws IOException {
     Note note = gson.fromJson(json, Note.class);
 //    note.setReplLoader(replLoader);
 //    note.jobListenerFactory = jobListenerFactory;
+    note.setLastModifiedUnixTime(content.getLastModifiedTime());
```

Comment:
Updating last modified time of the notebook here breaks plug-ability of storage layer because of all other storage layer implementation need to take care of it manually. 

Each paragraph has `dateCreated`, `dateStarted`, `dateFinished`, `dateUpdated`. I think there're already enough informations to know when notebook is updated at last.

So it's better remove `lastModifiedUnixTime` field and `setLastModifiedUnixTime()`  from `Note` class and make `getLastModifiedUnixTime()` iterates all paragraphs and get get last modified time of note based on `dateCreated`, `dateStarted`, `dateFinished`, `dateUpdated`.